### ISSUE TITLE
Add JSON Reporter

### ISSUE THREAD

**Issue by kehoecj** [Boeing/config-file-validator#11]
## Description

Add a JSON reporter type that will output the results of the config file validation in JSON format. This will be beneficial for projects that want to parse the output.

**Comment by gijsdb** [comment]
Do you have a preference on what format the JSON should be?

**Comment by kehoecj** [comment]
Hi @gijsdb - great question! Here are my initial thoughts but I'm open to alternative suggestions. I'd like the output on the CLI to be "pretty-print" style formatted if possible, for example like the output of `docker inspect [container]` but also easily parsed by tools like `jq`. For output I was thinking something like this:

```json
{
  "files": [
    {"path": "/path/to/config.yaml", "status": "passed"},
    {"path": "/path/to/config.json", "status":  "failed"}
  ],
  "summary": {
    "passed": 1,
    "failed": 1
  }
}
```

**Comment by ravsii** [comment]
Summary per file/config could be useful as well, maybe?

**Comment by gijsdb** [comment]
I am keen to have a go at this issue. Please assign me. 

@ravsii What do you mean with summary per file? Like what went wrong for a specific file?

**Comment by kehoecj** [comment]
> Summary per file/config could be useful as well, maybe?

Can you show 

> I am keen to have a go at this issue. Please assign me.

Thank you @gijsdb! Let me know if you have any questions during implementation

**Comment by ravsii** [comment]
@gijsdb Well, initially I meant exactly that, but then realised that the validator reports only the first error and stops, so yeah it doesn't make sense.

Although including the error probably would be useful.
```json
{
  "files": [
    {"path": "/path/to/config.yaml", "status": "passed"}, 
    {"path": "/path/to/config.json", "status":  "failed", "error": "error: yaml: line 3: did not find expected '-' indicator"}
  ],
  "summary": {
    "passed": 1,
    "failed": 1
  }
}
```

**Comment by kehoecj** [comment]
> @gijsdb Well, initially I meant exactly that, but then realised that the validator reports only the first error and stops, so yeah it doesn't make sense.
> 
> Although including the error probably would be useful.
> 
> ```json
> {
>   "files": [
>     {"path": "/path/to/config.yaml", "status": "passed"}, 
>     {"path": "/path/to/config.json", "status":  "failed", "error": "error: json: line 3: did not find expected '-' indicator"}
>   ],
>   "summary": {
>     "passed": 1,
>     "failed": 1
>   }
> }
> ```

Agree with adding the error

**Comment by gijsdb** [comment]
@kehoecj Two questions

Would you like a command line argument to specify if the output should be printed in JSON?
I assume you want the JSON reporter to print to stdout and not to a file?

**Comment by kehoecj** [comment]
> @kehoecj Two questions
> 
> Would you like a command line argument to specify if the output should be printed in JSON? 

Yes, I'm thinking something like `--output json` or `--reporter json`  

>I assume you want the JSON reporter to print to stdout and not to a file?

Yes, stdout for now although file could definitely be something to add down the line
